### ISSUE TITLE
Print full path of nextflow script in log

### ISSUE THREAD

**Issue by grst** [nextflow-io/nextflow#1820]
### Expected behavior and actual behavior

Actual: If nextflow encounters a compile-time-error it prints `Check script 'main.nf' at line X`

Expected: print the full path of the file. 

Context: In nf-core DSL2 pipelines, there are many many scripts called `main.nf` (one for each module). 

### Steps to reproduce the problem

```nextflow
#!/usr/bin/env nextflow

nextflow.enable.dsl = 2

process test {
    tag doesnotexist

    exec:
    println "Hello"
}

workflow { 
    test()
}
```

### Program output 

```console
N E X T F L O W  ~  version 20.10.0
Launching `./main.nf` [cranky_cori] - revision: 2c5587e4e4
No such variable: doesnotexist

 -- Check script 'main.nf' at line: 6 or see '.nextflow.log' file for more details
```

### Environment 

* Nextflow version: 20.10.0
* Java version: openjdk version "11.0.8-internal" 2020-07-14
* Operating system: Linux
* Bash version: 5.0.18

**Comment by abhi18av** [comment]
Hi @grst ,

I was looking into this request and I realized that the following code does isolate the file URI for the log message.

https://github.com/nextflow-io/nextflow/blob/c4b3617611fdbb9371956b13ff72f7d65b5af1e1/modules/nextflow/src/main/groovy/nextflow/util/LoggerHelper.groovy#L574 


However, there's remaining part of the aforementioned code does the processing w.r.t the current working directory.

```groovy
if( script.startsWith(pwd) ) {
    script = script.substring(pwd.length())
    while( script.startsWith('/') )
        script = script.substring(1)
}
```

For example, invoking the `1820.nf` script from the `$projectDir`prints out the relative path

```
base) Abhinavs-MacBook-Pro:nextflow eklavya$ nextflow _scratch/1820_issue/1820.nf
N E X T F L O W  ~  version 20.10.0
Launching `_scratch/1820_issue/1820.nf` [shrivelled_pare] - revision: 2c5587e4e4
No such variable: doesnotexist

 -- Check script '_scratch/1820_issue/1820.nf' at line: 6 or see '.nextflow.log' file for more details

```


Here's the file structure 

```
_scratch/
├── 1820_issue
│   ├── 1820.nf


```

So in conclusion, there's some optimization to facilitate the path printing behavior. 

I guess, what you're suggesting is to simply _always_ print the full path right?

**Comment by grst** [comment]
> I guess, what you're suggesting is to simply always print the full path right?

I don't see any downsides of that. Explicit is better than implicit.

**Comment by abhi18av** [comment]
Yeah it makes sense given that based on DSL2,  it's not necessary that the modules are even in the same directory.

I've initiated a draft PR https://github.com/nextflow-io/nextflow/pull/1903, once we've had a review - we can decide on merging/closing.

**Comment by pditommaso** [comment]
It's already reporting the file name relative to the module e.g. 

```
 Check script './modules/./index.nf' at line: 3 or see '.nextflow.log' file for more details
```


I think should be enough to resolve ambiguities.
